CMSgov/hcgov-design-system
# [MEA-906] Fix typescript errors when building

New design-system typescript components were not compiling correctly because hcgov had not been set up yet to handle typescript components.

Added:
- tsconfig.json 
- typescript: true in cmsds.config.js
- @types/react in package.json

## To Test
- First run on master to see error
- Build master of `design-system`
- Change package.json to point to master of `design-system`
```
    "@cmsgov/design-system": "file:../design-system/packages/design-system",
    "@cmsgov/design-system-docs": "file:../design-system/packages/design-system-docs",
    "@cmsgov/design-system-scripts": "file:../design-system/packages/design-system-scripts",
```
- Run `rm -rf node_modules && yarn install && yarn build`
- You should see errors like below: 
- Pull down this branch and run steps above
- No errors and running `yarn start` you should see props show on the Accordion page
